src/main/java/com/linecorp/armeria/client/routing/ServerGroupRegistry.java

```diff
@@ -0,0 +1,102 @@
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A memory server group register center.
+ */
+public class ServerGroupRegistry {
+    private static final Map<String, NodeSelector> serverGroups = new ConcurrentHashMap<>();
+
+    /**
+     * Register the server group.if the groupName has already registered,throw
+     * {@link ServerGroupRegisteredException}.
+     */
+    public static boolean register(String groupName, ServerGroup serverGroup,
+                                   NodeSelectStrategy nodeSelectStrategy) {
+        requireNonNull(groupName, "groupName");
+        requireNonNull(serverGroup, "serverGroup");
+        requireNonNull(nodeSelectStrategy, "nodeSelectStrategy");
+
+        NodeSelector existSelector = serverGroups.putIfAbsent(groupName,
+                nodeSelectStrategy.newSelector(serverGroup));
+        if (existSelector != null) {
+            throw new ServerGroupRegisteredException("Registry failed!Because of the group:" +
+                    groupName + " has registered already.");
+        }
+
+        return existSelector == null;
+    }
+
+    /**
+     * Replace the exist server group registry. if the groupName not registered yet
+     * throw {@link ServerGroupRegisteredException}.
+     */
+    public static boolean replace(String groupName, ServerGroup serverGroup,
+                                  NodeSelectStrategy nodeSelectStrategy) {
+        requireNonNull(groupName, "groupName");
+        requireNonNull(serverGroup, "serverGroup");
+        requireNonNull(nodeSelectStrategy, "nodeSelectStrategy");
+
+        NodeSelector existSelector = serverGroups.replace(groupName,
+                nodeSelectStrategy.newSelector(serverGroup));
+        if (existSelector == null) {
+            throw new ServerGroupRegisteredException("Replace failed!Because of the group:" +
+                    groupName + " not registered yet.");
+        }
+        return existSelector != null;
+    }
+
+    /**
+     * Get a {@link NodeSelector} by a groupName,or null if the groupName hasn't registered yet.
+     */
+    public static NodeSelector getNodeSelector(String groupName) {
+        requireNonNull(groupName, "groupName");
+
+        return serverGroups.get(groupName);
+    }
+
+    /**
+     * Get a {@link ServerGroup} by groupName,or null if the server group hasn't registered yet.
+     */
+    public static ServerGroup get(String groupName) {
+        requireNonNull(groupName, "groupName");
+
+        NodeSelector nodeSelector = serverGroups.get(groupName);
+        if (nodeSelector == null) {
+            return null;
+        }
+        return nodeSelector.serverGroup();
+    }
+
+    /**
+     * Select a node from the target server group.
+     */
+    public static Node selectNode(String groupName) {
+        NodeSelector nodeSelector = ServerGroupRegistry.getNodeSelector(groupName);
+        requireNonNull(nodeSelector, "The server group:" + groupName + " not registered yet.");
```

Comment:
- Could you raise a `ServerGroupRegisteredException` instead of NPE here?
- `"non-existent ServerGroup: " + groupName`